src/Agent.Worker/Handlers/Handler.cs

```diff
@@ -0,0 +1,100 @@
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Worker;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.Handlers
+{
+    public interface IHandler : IAgentService
+    {
+        IExecutionContext ExecutionContext { get; set; }
+        Dictionary<string, string> Inputs { get; set; }
+        string TaskDirectory { get; set; }
+
+        Task RunAsync();
+    }
+
+    public abstract class Handler : AgentService
```

Comment:
This is the base class for handlers. It has the common methods for the things the new handlers do: add endpoints/inputs/variables to the environment variable dictionary that gets passed to the process.